vis_services/lib/paper_network.py

```diff
@@ -105,46 +105,46 @@ def augment_graph_data(data, max_groups):
 
     #enhance the information that will be in the json handed off to d3
     for x in summary_graph.nodes():
-        summary_graph.node[x]["total_citations"] = sum([G.node[paper].get("citation_count", 0) for paper in G.nodes() if G.node[paper]["group"] == x])
-        summary_graph.node[x]["total_reads"] = sum([G.node[paper].get("read_count", 0) for paper in G.nodes() if G.node[paper]["group"] == x])
-        papers = sorted([G.node[paper] for paper in G.nodes() if G.node[paper]["group"] == x], key = lambda x: x.get("nodeWeight", 0), reverse = True)
+        summary_graph.nodes[x]["total_citations"] = sum([G.nodes[paper].get("citation_count", 0) for paper in G.nodes() if G.nodes[paper]["group"] == x])
+        summary_graph.nodes[x]["total_reads"] = sum([G.nodes[paper].get("read_count", 0) for paper in G.nodes() if G.nodes[paper]["group"] == x])
+        papers = sorted([G.nodes[paper] for paper in G.nodes() if G.nodes[paper]["group"] == x], key = lambda x: x.get("nodeWeight", 0), reverse = True)
         titles[x] = [p["title"]for p in papers]
-        summary_graph.node[x]["paper_count"] = len(papers)
+        summary_graph.nodes[x]["paper_count"] = len(papers)
 
     #attaching title 'word clouds' to the nodes
     significant_words = tf_idf.get_tf_idf_vals(titles)
-    for x in summary_graph.nodes():
+    for x in list(summary_graph.nodes()):
         #remove the ones with only 1 paper
-        if summary_graph.node[x]["paper_count"] == 1:
+        if summary_graph.nodes[x]["paper_count"] == 1:
             summary_graph.remove_node(x)
         else:
             #otherwise, give them a title
             #how many words should we show on the group? max 6, otherwise 1 per every 2 papers
-            summary_graph.node[x]["node_label"] =  dict(sorted(significant_words[x].items(), key = lambda x: x[1], reverse = True)[:6])
+            summary_graph.nodes[x]["node_label"] =  dict(sorted(list(significant_words[x].items()), key = lambda x: x[1], reverse = True)[:6])
 
 
     #remove all but top n groups from summary graph
     #where top n is measured by total citations from a group
     top_nodes = sorted([n for n in summary_graph.nodes(data = True)], key= lambda x : x[1]["total_citations"], reverse = True )[:max_groups]
-    top_nodes = [t for t in top_nodes if t >=1]
+    top_nodes = [t for t in top_nodes if t[0] >=1]
     top_node_ids = [n[0] for n in top_nodes]
-    for group_id in summary_graph.nodes():
+    for group_id in list(summary_graph.nodes()):
```

Comment:
Do we need a `list` here?

Reply:
list function is required because summary_graph is modified in the loop